### ISSUE TITLE
Unable to query a redshift cluster in github actions

### ISSUE THREAD

**Issue by raphaelvignes** [aws/amazon-redshift-python-driver#110]
## Driver version
2.0.907

## Redshift version

PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.38698\x00\x00

## Client Operating System
Linux (Ubuntu)

## Python version
3.10

## Table schema
default

## Problem description
Hello, we are facing issues using the python driver for Redshift. We are able to locally connect & run queries for integration tests on our cluster as such: 
```REDSHIFT_CONNEXION = redshift_connector.connect(
    user=USER,
    database=DATABASE,
    host=HOST,
    cluster_identifier=CLUSTER,
    password=PASSWORD,
    region=REGION,
    ssl=False,
)
...
    with REDSHIFT_CONNEXION.cursor() as curs:
        curs.execute(f'DROP TABLE IF EXISTS {case_id}')
```

But we are not able to perfom the same execution using github actions, with such context: 
```runs-on: ubuntu-20.04
    strategy:
      fail-fast: true
      matrix:
        python-version: ['3.10']
...
    - name: Test
      env:
        REDSHIFT_PASSWORD: '${{ secrets.REDSHIFT_PASSWORD }}'
      run: make test
```
-> This make command runs: `python -m pytest --junitxml=test-report.xml --cov=weaverbird --cov-report xml -s`

(full ci config file [here](https://github.com/ToucanToco/weaverbird/blob/master/.github/workflows/ci-server.yml) )

We notice that all our test case are failling when trying to run the `cursor.execute()` instruction. (But the `connect()` instruction seems to work) -> we have timeouts each time the ci tries to run the execute statement.

For the complete stack trace in github actions see: [weaverbird](https://github.com/ToucanToco/weaverbird/runs/6968527337?check_suite_focus=true)

**Comment by Brooke-white** [comment]
Hey @raphaelvignes , thanks for opening this issue. Tracing through the exception, I see `ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2396)` at the root. Was this previously working in your CI?

**Comment by raphaelvignes** [comment]
Hi @Brooke-white, we tried to connect with & without 
`ssl=False` it didn't work in both cases. 
We don't use a proxy locally nor in github actions. 
Python dependencies are the same locally and in CI (those in the `pyproject.toml` file). 
As for the os dependencies, we didn't manage to replicate the issue using github actions' `ubuntu:latest` image.

**Comment by Brooke-white** [comment]
Hey @raphaelvignes, 

I looked at the [redshift tests](https://github.com/ToucanToco/weaverbird/blob/a513bc6cd215b667f4c1a0511bc3d338b7ae94bf/server/tests/backends/sql_translator/test_sql_redshift_translator_steps.py), as well as the CI log a bit more and am wondering if this issue could have to do with database locks.

As the CI **is able to connect**, but sees issue with this [drop table stmt](https://github.com/ToucanToco/weaverbird/blob/a513bc6cd215b667f4c1a0511bc3d338b7ae94bf/server/tests/backends/sql_translator/test_sql_redshift_translator_steps.py#L87), I'm thinking this could be worth looking into.

I've seen issues like this before when contributing to sqlalchemy-redshift, where I wrote some tests that worked locally but failed in CI, and ended up determining it had something to do with the tests being run in parallel mode on CI resulting in locks on the database (I was trying to create and drop tables & schemas). Not sure if this could apply to your situation but thought I'd mention.

Maybe one simple test could be to just try to execute some simple `select 1` statement when using something like [netkos/act](https://github.com/nektos/act) and then using your CI.

And speaking of locks, it'd be worthwhile to take a look at the [Redshift system tables related to database locks](https://docs.aws.amazon.com/redshift/latest/dg/r_STV_LOCKS.html) to get a clear idea if that's coming into play here. It looks like you folks are using a dedicated Redshift cluster, so this should be easy to check locally using a tool like redshift-connector, Redshift Query Console or your preferred query tool.
